### PR TITLE
STREAMS-935 Add description of custom metrics available for monitoring

### PR DESCRIPTION
**Pull request by cremond**
Thank you for your contribution to the Axway Streams documentation.

## Describe the changes

I have added a description of the metrics provided by Streams so that users can monitor their installations

## Checklist for contributors

Before submitting this PR, please make sure:

* [x] You have read the [contribution guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/)
* [x] You have signed the [Axway CLA](https://cla.axway.com/)
* [x] You have verified the technical accuracy of your change
* [x] You have verified that your change does not expose any sensitive information (passwords, keys, etc.)
* [x] You have followed the [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)  (unless this is is a Netlify CMS contribution)

_Put an x in the boxes that apply. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your change._

## Checklist for approvers

_This section is to be filled out by project maintainers only._

Before approving this PR, please make sure it:

* [x] Does not have conflicts with the base branch
* [x] Is believed to be accurate (technical accuracy to be checked with an SME for PRs originating from outside R&D)
* [x] Follows [Axway style](https://techweb.axway.com/confluence/display/RIE/Style+guide)
* [x] Follows [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)
* [x] Follows [best practices](https://axway-open-docs.netlify.com/docs/contribution_guidelines/bestpracticedevdoc/)
* [x] Does not contain typos, grammatical errors, etc.
* [x] Does not expose any sensitive information
* [x] Passes the Markdown linter rules (automated via CI check)
* [x] Does not contain broken links

### CODE REVIEW

File: content/en/docs/metrics.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+## Introduction
+
+Streams provides metrics in a Prometheus format when the
+[monitoring parameters](/docs/install/helm-parameters/#monitoring-parameters) are enabled.
+These are meant to be scraped by a Prometheus server and visualized through Grafana on our custom dashboards.
+
+## Metrics types
+
+Although we assume your are familiare with the Prometheus metrics format, here is a quick explanation of the main
```
Comment (allandenis38) [inline content/en/docs/metrics.md:17]:
```suggestion
Although we assume your are familiar with the Prometheus metrics format, here is a quick explanation of the main
```

File: content/en/docs/metrics.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+metrics types produced by [micrometer](https://micrometer.io/docs/registry/prometheus) and how to make use of them :
```
Comment (allandenis38) [inline content/en/docs/metrics.md:18]:
```suggestion
metrics types produced by [micrometer](https://micrometer.io/docs/registry/prometheus) and how to make use of them:
```

File: content/en/docs/metrics.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+## Metrics
+
+### Custom metrics
+
+We have implemented the following custom metrics. All of those have labels describing the service name, type,
+and unique identifier, as well as some custom labels when suitable.
+
+| Type    | Metric                                  | Description                                               | Example usage |
+| -----   | --------------------------              | ---------------------------                               | ------        |
+| Gauge   | streams_global_topics                   | Number of topics in total on Streams                      |  |
```
Comment (allandenis38) [inline content/en/docs/metrics.md:38]:
```suggestion
| Gauge   | streams_global_topics                   | Total number of Streams topics                       |  |
```

File: content/en/docs/metrics.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+| Gauge   | streams_global_persistent_subscriptions | Number of persistent subscriptions in total on Streams    | streams_global_persistent_subscriptions{subscription_status="active"} |
```
Comment (allandenis38) [inline content/en/docs/metrics.md:39]:
```suggestion
| Gauge   | streams_global_persistent_subscriptions | Total number of persistent subscriptions    | streams_global_persistent_subscriptions{subscription_status="active"} |
```

File: content/en/docs/metrics.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+| Gauge   | streams_topics                          | Number of active topics on a Streams publisher            | streams_topics{streams_service="publisher-http-poller"} |
+| Gauge   | streams_active_subscriptions            | Number of active subscriptions on a Streams subscriber    | streams_topics{streams_service="subscriber-sse"} |
+| Counter | streams_input_events_total              | Number of messages received by a Streams service          | rate(streams_input_events_total{streams_name="streams-subscriber-sse"}[2m]) |
+| Counter | streams_output_events_total             | Number of messages emitted by a Streams service           | rate(streams_input_events_total{data_type="patch"}[2m]) |
+
+The keyword global here refers to the fact that some metrics should have the same value everywhere,
```
Comment (allandenis38) [inline content/en/docs/metrics.md:45]:
```suggestion
The keyword global refers to the scope of the metrics. Global metrics are reported by all instances of the same type of service and will have the same value regardless of the service instance that reports them,
```

File: content/en/docs/metrics.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+whereas the others only give information about the service which reported them.
```
Comment (allandenis38) [inline content/en/docs/metrics.md:46]:
```suggestion
whereas the others only give information about the service instance which reported them.
```

**Review by allandenis38 — CHANGES_REQUESTED** [review]

**Comment by allandenis38** [comment]
Hi @lbadenhop, you can review this MR. Don't hesitate to make changes on my suggestions.

File: content/en/docs/metrics.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+Although we assume your are familiare with the Prometheus metrics format, here is a quick explanation of the main
+metrics types produced by [micrometer](https://micrometer.io/docs/registry/prometheus) and how to make use of them :
+
+* Gauge: A gauge is the simplest metric to use. Like the fuel gauge of a car, it is a value that goes up and down
+and has immediate meaning.
+
+* Counter: The more common metric type in Prometheus, a counter is a number that _always goes up_ (that restriction makes
+it more reliable). It is usually differentiated into a rate to get an indication of how fast that number is going up.
+
+* Timer : A timer records tasks durations by keeping a counter of completed tasks, and a counter of their cumulated time.
```
Comment (lbadenhop) [inline content/en/docs/metrics.md:26]:
```suggestion
* Timer: A timer records task durations by keeping a counter of completed tasks, and a counter of their cumulated time.
```

File: content/en/docs/metrics.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+Utilize those metrics through ratios and rates.
+
+## Metrics
+
+### Custom metrics
+
+We have implemented the following custom metrics. All of those have labels describing the service name, type,
```
Comment (lbadenhop) [inline content/en/docs/metrics.md:33]:
```suggestion
The following custom metrics have been implemented. They include labels describing the service name, type,
```

File: content/en/docs/metrics.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+| Gauge   | streams_active_subscriptions            | Number of active subscriptions on a Streams subscriber    | streams_topics{streams_service="subscriber-sse"} |
+| Counter | streams_input_events_total              | Number of messages received by a Streams service          | rate(streams_input_events_total{streams_name="streams-subscriber-sse"}[2m]) |
+| Counter | streams_output_events_total             | Number of messages emitted by a Streams service           | rate(streams_input_events_total{data_type="patch"}[2m]) |
+
+The keyword global here refers to the fact that some metrics should have the same value everywhere,
+whereas the others only give information about the service which reported them.
+
+### Default Spring Boot metrics
+
+Our frameworks provide a lot of additional metrics. However,
```
Comment (lbadenhop) [inline content/en/docs/metrics.md:50]:
```suggestion
Our frameworks provides additional metrics; however,
```

File: content/en/docs/metrics.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+we recommend you pay attention to the following timers in particular.
+
+* http\_client\_request\_seconds : timer of the web requests _emitted_ by Streams (i.e. polling or webhook).
```
Comment (lbadenhop) [inline content/en/docs/metrics.md:53]:
```suggestion
* http\_client\_request\_seconds : timer of the web requests _emitted_ by Streams (i.e., polling or webhook).
```

File: content/en/docs/metrics.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+* http\_server\_requests\_seconds : timer of the web requests _received_ and handled by Streams (i.e. REST api, self health-check, and http-post).
```
Comment (lbadenhop) [inline content/en/docs/metrics.md:55]:
```suggestion
* http\_server\_requests\_seconds : timer of the web requests _received_ and handled by Streams (i.e., REST API, self health-check, and http-post).
```

File: content/en/docs/metrics.md [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+
+These are split into a sum of cumulated seconds spent waiting for the requests,
+a counter of the number of the number of requests, and here a gauge of the maximum request time over the latest period of time.
```
Comment (lbadenhop) [inline content/en/docs/metrics.md:58]:
```suggestion
a counter of the number of requests, and a gauge of the maximum request time over the latest period of time.
```

File: content/en/docs/metrics.md [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+For instance, to get the average incoming requests processing time, you could plot :
```
Comment (lbadenhop) [inline content/en/docs/metrics.md:59]:
```suggestion
For example, to get the average incoming requests' processing time, you could plot :
```

File: content/en/docs/metrics.md [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+| Counter | streams_input_events_total              | Number of messages received by a Streams service          | rate(streams_input_events_total{streams_name="streams-subscriber-sse"}[2m]) |
+| Counter | streams_output_events_total             | Number of messages emitted by a Streams service           | rate(streams_input_events_total{data_type="patch"}[2m]) |
+
+The keyword global here refers to the fact that some metrics should have the same value everywhere,
+whereas the others only give information about the service which reported them.
+
+### Default Spring Boot metrics
+
+Our frameworks provide a lot of additional metrics. However,
+we recommend you pay attention to the following timers in particular.
```
Comment (lbadenhop) [inline content/en/docs/metrics.md:51]:
```suggestion
we recommend you pay attention to the following timers:
```

**Review by lbadenhop — APPROVED** [review]

**Review by allandenis38 — APPROVED** [review]

**Pull request merged**